src/sentry/rules/conditions/event_attribute.py

```diff
@@ -189,21 +192,14 @@ def render_label(self) -> str:
         }
         return self.label.format(**data)
 
-    def passes(self, event: GroupEvent, state: EventState, **kwargs: Any) -> bool:
-        attr = self.get_option("attribute")
+    def _passes(self, attribute_values: Sequence[Any]) -> bool:
         match = self.get_option("match")
         value = self.get_option("value")
 
-        if not (attr and match and value):
+        if not (match and (value or match in (MatchType.IS_SET, MatchType.NOT_SET))):
```

Comment:
is `value` always `None` with the `is_set` and `is_not_set` matchers? maybe add a second test for one of them.

Reply:
I would maybe separate out the logic here to make it a bit cleaner to read//less likely to get misinterpreted
`not ((match and value) or (match in (MatchType.IS_SET, MatchType.NOT_SET)))`